src/backends/backend/triton_model_instance.cc

```diff
@@ -128,6 +128,264 @@ TritonModelInstance::CreateInstance(
   return Status::Success;
 }
 
+Status
+TritonModelInstance::Schedule(
+    std::vector<std::unique_ptr<InferenceRequest>>&& requests)
+{
+  // Use a thread local vector to avoid needing to malloc each
+  // time an inference is run.
+  thread_local std::vector<TRITONBACKEND_Request*> triton_requests(1024);
```

Comment:
Note that this may need to change later on when multiple instances share the same thread, or at least need to be documented in `TRITONBACKEND_ModelInstanceExecute()` that the `requests` may be overwritten once the function returns. For instance, in TRT the requests are released in the `ProcessResponse()` and that may happen after the next batch is scheduled, so instead of passing `requests` to the `ProcessResponse()`, the request handles in `requests` should be copied.

Reply:
Thanks for the insight! I will keep that in mind for DEVICE_BLOCKING implementation.